controllers/helmrelease_controller.go

```diff
@@ -704,7 +704,15 @@ func (r *HelmReleaseReconciler) handleHelmActionResult(hr *v2.HelmRelease, revis
 	}
 }
 
-func (r *HelmReleaseReconciler) helmReleasesForHelmChart(obj handler.MapObject) []reconcile.Request {
+func (r *HelmReleaseReconciler) enqueueForHelmChartArtifactChange(obj handler.MapObject) []reconcile.Request {
+	hc, ok := obj.Object.(*sourcev1.HelmChart)
+	if !ok {
+		panic(fmt.Sprintf("Expected a HelmChart, got %T", hc))
```

Comment:
Can we return an error instead of crashing the process?

Reply:
The reason I use a panic here is because it should never happen. If it happens, something is wired wrong (by us) and logging this does not make this obvious.